docs/source/coder/globalKeys.rst

```diff
@@ -0,0 +1,281 @@
+create the dummy function:
+::
+
+    >>> def myfunc2(*args, **kwargs):
+    ...     pass
+    ...
+
+Next, compile some positional and keyword arguments and a custom name for this
+event. Positional arguments must be passed as tuples, and keyword arguments as
+dictionaries.
+::
+
+    >>> args = (1, 2)
+    >>> kwargs = dict(foo=3, bar=4)
+    >>> name = 'my name'
+
+Finally, specify the key and a combination of modifiers. While key names are
+just strings, modifiers are lists or tuples of modifier names.
+::
+
+    >>> key = 'b'
+    >>> modifiers = ['ctrl', 'alt']
+
+.. note::
+
+   Even when specifying only a single modifier key, `modifiers` must be a list
+   or tuple, e.g., `modifiers = ['ctrl']`.
+
+We are now ready to create the global event key.
+::
+
+    >>> event.globalKeys.add(key=key, modifiers=modifiers,
+    ... func=myfunc2, func_args=args, func_kwargs=kwargs,
+    ... name=name)
+
+Check that the global event key was successfully added.
+::
+
+    >>> event.globalKeys
+    <_GlobalEventKeys :
+        [A] -> 'myfunc' <function myfunc at 0x10669ba28>
+        [CTRL] + [ALT] + [B] -> 'my name' <function myfunc2 at 0x112eecb90>
+    >
+
+The key combination `[CTRL] + [ALT] + [B]` is now associated with the function
+`myfunc2`, which will be called in the following way:
+::
+
+    myfunc2(1, 2, foo=2, bar=4)
+
+.. _indexing:
+
+Indexing
+--------
+
+`event.globalKeys` can be accessed like an ordinary dictionary. The index keys
+are `(key, modifiers)` namedtuples.
+::
+
+    >>> event.globalKeys.keys()
+    [_IndexKey(key='a', modifiers=()), _IndexKey(key='b', modifiers=('ctrl', 'alt'))]
+
+To access the global event associated with the key combination
+`[CTRL] + [ALT] + [B]`, we can do
+
+    >>> event.globalKeys['b', ['ctrl', 'alt']]
+    _GlobalEvent(func=<function myfunc2 at 0x112eecb90>, func_args=(1, 2), func_kwargs={'foo': 3, 'bar': 4}, name='my name')
+
+To make access more convenient, specifying the modifiers is optional in case
+none were passed to :func:`psychopy.event.globalKeys.add` when the global
+event key was added, meaning the following commands are identical.
+::
+
+    >>> event.globalKeys['a', ()]
+    _GlobalEvent(func=<function myfunc at 0x10669ba28>, func_args=(), func_kwargs={}, name='myfunc')
+    >>> event.globalKeys['a']
+    _GlobalEvent(func=<function myfunc at 0x10669ba28>, func_args=(), func_kwargs={}, name='myfunc')
+
+All elements of a global event can be accessed directly.
+::
+
+    >>> index = ('b', ['ctrl', 'alt'])
+    >>> event.globalKeys[index].func
+    <function myfunc2 at 0x112eecb90>
+    >>> event.globalKeys[index].func_args
+    (1, 2)
+    >>> event.globalKeys[index].func_kwargs
+    {'foo': 3, 'bar': 4}
+    >>> event.globalKeys[index].name
+    'my name'
+
+Removing global event keys
+--------------------------
+There are three ways to remove global event keys:
+
+- using :func:`psychopy.event.globalKeys.remove`,
+- using `del`, and
+- using :func:`psychopy.event.globalKeys.clear`.
```

Comment:
Good thoughts! About "how thoroughly" this is is implemented: it's just a matter of subclassing from the right object types!

Regarding `remove`:
I had the same rationale: we do have `add`, so probably we should also have `remove`. Another reason to keep `remove` is that it can be used to remove all global event keys at once, via `remove('all')` -- I forgot to mention this in the docs! This would not be easily possible via `pop` or `del`. But I propose we can safely drop `clear` in favor of `remove('all')`. I can add the `pop` approach to the docs as well as another approach to remove individual keys.